pkg/declared/resources.go

```diff
@@ -42,6 +42,8 @@ type Resources struct {
 	objectMap *orderedmap.OrderedMap[core.ID, *unstructured.Unstructured]
 	// commit of the source in which the resources were declared
 	commit string
+	// previousCommit is the preceding commit to the commit
+	previousCommit string
```

Comment:
The objectMap and commit are locked by the mutex. I think we need to do the same for the previousCommit to avoid race conditions. 

But looking at the code, I don't quite understand why it's using a setter and getter. Maybe whoever wrote it (maybe me) wanted to avoid touching all the other methods. 

I think the right way to fix this would be to remove the getter/setter, Lock the Update, and RLock all the other methods.

Reply:
We could keep the getter / setter to keep appearance clean and just extract the lock to the calling function level? My preference is to have separate getter/setter for each member but it could be just too much overhead.

Reply:
The reason they were put in the same setter is because they need to be updated together atomicly, in the same lock. Same goes for the previousCommit.

And now with previousCommit, the read and write need to be in the same lock. So it doesn't make sense to have a setter.

Reply:
Okie, I've lifted the r/w lock out of the getter/setter.